src/Countries/Ghana.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Ghana extends Country
+{
+
+    protected string $timezone = 'Africa/Accra';
+
+    public function countryCode(): string
+    {
+        return 'gh';
+    }
+
+    /**
+     * Return carbon date for christmas
+     * @return CarbonImmutable
+     */
+    protected function getChristmasDay(int $year)
+    {
+        return  new CarbonImmutable($year . "-12-25");
+    }
+
+    /**
+     * @return array<string, CarbonImmutable>
+     */
+    protected function christmasDay(int $year): array
+    {
+        $christmasDay = $this->getChristmasDay($year);
+        $key = 'Christmas Day';
+
+        if ($christmasDay->isSaturday()) {
+            $key .= ' (substitute day)';
+            $christmasDay = $christmasDay->next('monday');
+        }
+
+        if ($christmasDay->isSunday()) {
+            $key .= ' (substitute day)';
+            $christmasDay = $christmasDay->next('tuesday');
+        }
+
+        return [$key => $christmasDay];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable>
+     */
+    protected function boxingDay(int $year): array
+    {
+        $christmasDay = $this->getChristmasDay($year);
+        $boxingDay = new CarbonImmutable($year . "-12-26");
+        $key = 'Boxing Day';
+
+        if ($christmasDay->isFriday()) {
+            $key .= ' (substitute day)';
+            $boxingDay = $boxingDay->next('monday');
+        }
+
+        if ($christmasDay->isSaturday()) {
+            $key .= ' (substitute day)';
+            $boxingDay = $boxingDay->next('tuesday');
+        }
+
+        return [$key => $boxingDay];
+    }
+
+    /**
+     * Get holiday
+     * 
+     * If it falls on a weekend, the new day to be observed is the next monday
+     * 
+     * @return array<string, CarbonImmutable>
+     *
+     */
+    protected function getHoliday(string $nameOfHoliday, int $year, string $monthAndDay): array
+    {
+        $newYearsDay = new CarbonImmutable($year . "-" . $monthAndDay, $this->timezone);
+        $key = $nameOfHoliday;
+
+        if ($newYearsDay->isWeekend()) {
+            $key .= ' (substitute day)';
+            $newYearsDay = $newYearsDay->next('monday');
+        }
+
+        return [$key => $newYearsDay];
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge(
+            $this->getHoliday('New Year Day', $year, "01-01"),
```

Comment:
Can you make this more in line with how other Country classes do this? You can have a look at Belgium.php for example.

Reply:
Sure. Code has been updated